### PR TITLE
Scaffolding to get started with SIMD on the Vector API

### PR DESCRIPTION
**Pull request by andreaTP**
This is one possible setup to start implementing SIMD with the Vector API.
There are a few tricks to make it working (project setup, a sprinkle of reflection etc.) I'm happy to turn this around when a different approach is proposed :slightly_smiling_face:

### CODE REVIEW

**Comment by electrum** [comment]
This looks cool! I hope to have time to review this weekend.

File: runtime-tests/src/test/java/com/dylibso/chicory/testing/TestModule.java [hunk 1]
```diff
@@ -60,8 +64,27 @@ public TestModule(Module module) {
         this.module = module;
     }
 
+    private static Map<OpCode, Machine.OpImpl> additionalOpcodes;
+
+    static {
+        // TODO: discuss - either we use reflection or we need to restructure the project modules
```
Comment (andreaTP) [inline runtime-tests/src/test/java/com/dylibso/chicory/testing/TestModule.java:70]: An alternative approach would be to fully duplicate the `runtime-tests` module, the downside is that needs a bit of fiddling with profiles in Maven.

**Comment by andreaTP** [comment]
self-reminder: follow up task to remove this: https://github.com/dylibso/chicory/blob/main/wasm/src/test/resources/wasm/simd_load.0.wasm

File: runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+import com.dylibso.chicory.runtime.exceptions.WASMRuntimeException;
 import com.dylibso.chicory.wasm.exceptions.ChicoryException;
+import com.dylibso.chicory.wasm.types.OpCode;
+import java.util.Map;
 
 public interface Machine {
 
     long[] call(int funcId, long[] args) throws ChicoryException;
+
+    Map<OpCode, OpImpl> additionalOpCodes();
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java:12]: > All of these changes are specific to the interpreter and don't make sense for AOT.

I haven't thought much about the compatibility with AOT indeed, I think I imagined invoking those as `callHost` or similar.

But I like the suggestion regarding `eval`, I'd go for it!

File: runtime-tests/pom.xml [hunk 3]
```diff
@@ -220,4 +220,228 @@
       </plugin>
     </plugins>
   </build>
+
+  <profiles>
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply runtime-tests/pom.xml:224]: I'll double-check while rebasing :+1:

**Comment by evacchi** [comment]
superseded by https://github.com/dylibso/chicory/pull/652

**Pull request closed without merging**